linera-core/src/chain_worker/state.rs

```diff
@@ -117,13 +117,9 @@ where
         &mut self,
         height: BlockHeight,
     ) -> Result<Option<Certificate>, WorkerError> {
-        self.ensure_is_active()?;
-        let certificate_hash = match self.chain.confirmed_log.get(height.try_into()?).await? {
-            Some(hash) => hash,
-            None => return Ok(None),
-        };
-        let certificate = self.storage.read_certificate(certificate_hash).await?;
-        Ok(Some(certificate))
+        ChainWorkerStateWithTemporaryChanges { state: self }
```

Comment:
nit: We may prefer `ChainWorkerStateWithTemporaryChanges(self)` if we're going to use this syntax a lot